pages/deployments/oidc/oidc_with_aws.md

```diff
@@ -0,0 +1,168 @@
+---
+keywords: oidc, authentication, IAM, roles, AWS
+---
+
+# OIDC with AWS
+
+The [Buildkite Agent's oidc command](/docs/agent/v3/cli-oidc) allows you to request an OIDC token representing the current job. These tokens can be exchanged on federated systems like AWS for an identity and access management (IAM) role with AWS-scoped permissions.
+
+See the [OpenID Connect Core documentation](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) for more information about how OIDC tokens are constructed and how to extract and use claims.
+
+See the [AWS OpenID Connect identity provider in IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) for more information on their implementation of the federated system.
+
+We will be using two Buildkite Plugins to utilise OIDC with AWS and our Pipelines.
+
+- Buildkite Plugin [aws-assume-role-with-web-identity-buildkite-plugin](https://github.com/buildkite-plugins/aws-assume-role-with-web-identity-buildkite-plugin)
+- Buildkite Plugin [aws-ssm-buildkite-plugin](https://github.com/buildkite-plugins/aws-ssm-buildkite-plugin)
+
+## Step 1: Setup OIDC provider in AWS account
+
+First we'll set up an AWS IAM OpenID Connect (OIDC) provider using the following configuration.
+
+  URL: `https://agent.buildkite.com`
+
+  Audience: `sts.amazonaws.com`
+
+## Step 2: Create a new (or update an existing) IAM role to use with your pipeline(s)
+
+In this example the IAM role is named: `compute-ssm-oidc-example`. When creating the role you must `Select Custom Trust Policy`.
```

Comment:
Any maybe in here we should also link to the AWS docs on creating an IAM role with a custom trust policy   https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-custom.html